src/System.Linq/tests/ConcatTests.cs

```diff
@@ -251,20 +251,34 @@ public void CountOfConcatIteratorShouldThrowExceptionOnIntegerOverflow()
             var supposedlyLargeCollection = new DelegateBasedCollection<int> { CountWorker = () => int.MaxValue };
             var tinyCollection = new DelegateBasedCollection<int> { CountWorker = () => 1 };
 
+            Action<Action> assertThrows = (testCode) =>
+            {
+                // The full .NET framework uses unsigned arithmetic summing up collection counts.
+                // See https://github.com/dotnet/corefx/pull/6131.
+                if (PlatformDetection.IsFullFramework)
+                {
+                    testCode();
+                }
+                else
+                {
+                    Assert.Throws<OverflowException>(testCode);
```

Comment:
That mainly seemed to be about changing OOMs into OverflowExceptions, but this change suggests that no exception (no OOM) happens in the desktop case, yet we're throwing an overflow exception in the core case?